lib/mumukit/bridge.rb

```diff
@@ -25,16 +25,28 @@ def initialize(test_runner_url)
       #    expectation_results: [{binding:string, inspection:string, result:symbol}],
       #    feedback: string}
       def run_tests!(request)
-        response = post_to_server(request)
-        response_type = ResponseType.for_response response
-        response_type.parse response
+        with_sever_response request, 'test' do |response|
+          response_type = ResponseType.for_response response
+          response_type.parse response
+        end
+      end
+
+      def run_query!(request)
+        with_sever_response request, 'query' do | it |
+          {status: it['exit'].to_sym, result: it['out']}
```

Comment:
However if you think there is something with common logic that can be extracted to another module, reused in both tasks of parsing a response, and that it makes code actually simpler, be my guest :stuck_out_tongue: